packages/cli/src/steps/downloadPreviewClient/downloadAndWriteClient.ts

```diff
@@ -0,0 +1,23 @@
+import axios from 'axios';
+import { WriteStream } from 'fs';
+import { getFS } from '../../dependencies';
+import { ZIP_PATH } from './constants';
+
+const fs = getFS();
+
+export async function downloadAndWriteClient(url): Promise<string> {
+  const writer = fs.createWriteStream(ZIP_PATH);
+  const res = await axios.get(url, {
+    responseType: 'stream',
+  });
+  res.data.pipe(writer);
```

Comment:
i guess its never an issue as the http request takes a tick, but you should always wait for the open event of the writer stream before piping

```
writer.on('open', () => res.data.pipe(writer));
```

Reply:
you might need to move waiting for the http request in front of 

```
const writer = fs.createWriteStream(ZIP_PATH);
```

in order to ad the event listner of the wirte stream in the same process.tick